Sources/FuturedKit/NavigationLinks.swift

```diff
@@ -0,0 +1,61 @@
+import SwiftUI
+///
+/// ViewModifier and protocol for creating and handling NavigationLinks
```

Comment:
I would describe here what problem it solves and why using basic `NavigationLink` is not sufficient.